src/Analyzer.Utilities/TemplateDiscovery.cs

```diff
@@ -108,43 +108,50 @@ public static IEnumerable<TemplateAndParams> FindParameterFilesForTemplate(FileI
         public static bool IsValidTemplate(FileInfo file)
         {
-            // Assume bicep files are valid, they are compiled/verified later
-            if (file.Extension.Equals(".bicep", StringComparison.OrdinalIgnoreCase))
+            try
             {
-                return true;
-            }
+                // Assume bicep files are valid, they are compiled/verified later
+                if (file.Extension.Equals(".bicep", StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
 
-            using var fileStream = new StreamReader(file.OpenRead());
-            var reader = new JsonTextReader(fileStream);
+                using var fileStream = new StreamReader(file.OpenRead());
+                var reader = new JsonTextReader(fileStream);
 
-            reader.Read();
-            if (reader.TokenType != JsonToken.StartObject)
-            {
-                return false;
-            }
+                reader.Read();
+                if (reader.TokenType != JsonToken.StartObject)
+                {
+                    return false;
+                }
 
-            while (reader.Read())
-            {
-                if (reader.Depth == 1 && reader.TokenType == JsonToken.PropertyName)
+                while (reader.Read())
                 {
-                    if (string.Equals((string)reader.Value, "$schema", StringComparison.OrdinalIgnoreCase))
+                    if (reader.Depth == 1 && reader.TokenType == JsonToken.PropertyName)
                     {
-                        reader.Read();
-                        if (reader.TokenType != JsonToken.String)
+                        if (string.Equals((string)reader.Value, "$schema", StringComparison.OrdinalIgnoreCase))
+                        {
+                            reader.Read();
+                            if (reader.TokenType != JsonToken.String)
+                            {
+                                return false;
+                            }
+
+                            return validSchemaRegex.IsMatch((string)reader.Value);
+                        }
+                        else if (!validTemplateProperties.Any(property => string.Equals((string)reader.Value, property, StringComparison.OrdinalIgnoreCase)))
                         {
                             return false;
                         }
-
-                        return validSchemaRegex.IsMatch((string)reader.Value);
-                    }
-                    else if (!validTemplateProperties.Any(property => string.Equals((string)reader.Value, property, StringComparison.OrdinalIgnoreCase)))
-                    {
-                        return false;
                     }
                 }
-            }
 
-            return false;
+                return false;
+            }
+            catch {
+                // if template fails to parse in any way then also consider invalid template
```

Comment:
```suggestion
                // If template fails to parse in any way then also consider invalid template
```
Since you have to make edits, then this also